Low level error dump when pushing existing directory

I'm getting the error below when trying to push to an already existing directory. Currently the directory `admin` in the server contains:

```
CensoEstGr2015_25601263.pdf  CensoEstPos2015_25601263.pdf
```

My localy directory contains:

```
admin/:
contratos  cv-1.pdf  cv.pdf  facultad

admin/contratos:
Balcarce 3c corto borrador.odt  Balcarce 5c corto.odt     Contrato Alberti.odt
Balcarce 3c largo.odt           Balcarce 5c prórroga.odt

admin/facultad:
inscripcion maestria  rematriculacion

admin/facultad/inscripcion maestria:
comprobante inscripcion analisis multivariado.pdf
comprobante inscripcion modelo lineal.pdf

admin/facultad/rematriculacion:
ActDatEstGr2014_25601263.pdf  ActDatEstPos2014_25601263.pdf
```

Now I run `drive push admin` and get:

```
Resolving...
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4d9db6]

goroutine 50 [running]:
github.com/odeke-em/drive/src.merge(0xc8200d6e40, 0xc8200d6de0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/tmp/go/src/github.com/odeke-em/drive/src/changes.go:508 +0x526
github.com/odeke-em/drive/src.(*Commands).resolveChangeListRecv(0xc820015040, 0xc820015cc0, 0x0, 0x0, 0x0, 0xd85808, 0x0, 0x0, 0x0, 0x0)
	/tmp/go/src/github.com/odeke-em/drive/src/changes.go:368 +0xa20
github.com/odeke-em/drive/src.(*Commands).changeSlice(0xc820015040, 0xc8200d6d80)
	/tmp/go/src/github.com/odeke-em/drive/src/changes.go:465 +0x314
created by github.com/odeke-em/drive/src.(*Commands).resolveChangeListRecv
	/tmp/go/src/github.com/odeke-em/drive/src/changes.go:419 +0x1072

goroutine 1 [runnable]:
sync.runtime_Semacquire(0xc8200e7c9c)
	/usr/lib/go/src/runtime/sema.go:43 +0x26
sync.(*WaitGroup).Wait(0xc8200e7c90)
	/usr/lib/go/src/sync/waitgroup.go:126 +0xb4
github.com/odeke-em/drive/src.(*Commands).resolveChangeListRecv(0xc820015040, 0xc8200b1728, 0x0, 0x0, 0x0, 0xd85808, 0x0, 0x0, 0x0, 0x0)
	/tmp/go/src/github.com/odeke-em/drive/src/changes.go:424 +0x10b3
github.com/odeke-em/drive/src.(*Commands).doChangeListRecv(0xc820015040, 0xc8200bfa80, 0x6, 0xc8200ec040, 0x1a, 0xc8204a6160, 0xc8204a6000, 0x1, 0x0, 0x0, ...)
	/tmp/go/src/github.com/odeke-em/drive/src/changes.go:191 +0x338
github.com/odeke-em/drive/src.(*Commands).byRemoteResolve(0xc820015040, 0xc8200bfa80, 0x6, 0xc8200ec040, 0x1a, 0xc8204a6000, 0x1, 0x0, 0x0, 0x0, ...)
	/tmp/go/src/github.com/odeke-em/drive/src/changes.go:139 +0x187
github.com/odeke-em/drive/src.(*Commands).changeListResolve(0xc820015040, 0xc8200bfa80, 0x6, 0xc8200ec040, 0x1a, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
	/tmp/go/src/github.com/odeke-em/drive/src/changes.go:156 +0x264
github.com/odeke-em/drive/src.(*Commands).Push(0xc820015040, 0x0, 0x0)
	/tmp/go/src/github.com/odeke-em/drive/src/push.go:62 +0x47a
main.(*pushCmd).Run(0xc8200d2aa0, 0xc82000a180, 0x9, 0x9, 0xc8200c7500)
	/tmp/go/src/github.com/odeke-em/drive/cmd/drive/main.go:798 +0x237
github.com/odeke-em/command.Run()
	/tmp/go/src/github.com/odeke-em/command/command.go:202 +0xaa
github.com/odeke-em/command.ParseAndRun()
	/tmp/go/src/github.com/odeke-em/command/command.go:209 +0x19
main.main()
	/tmp/go/src/github.com/odeke-em/drive/cmd/drive/main.go:122 +0x21ce

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/lib/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 5 [syscall]:
os/signal.loop()
	/usr/lib/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/lib/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 6 [select, locked to thread]:
runtime.gopark(0x9eac48, 0xc820030728, 0x914c48, 0x6, 0x440e18, 0x2)
	/usr/lib/go/src/runtime/proc.go:185 +0x163
runtime.selectgoImpl(0xc820030728, 0x0, 0x18)
	/usr/lib/go/src/runtime/select.go:392 +0xa64
runtime.selectgo(0xc820030728)
	/usr/lib/go/src/runtime/select.go:212 +0x12
runtime.ensureSigM.func1()
	/usr/lib/go/src/runtime/signal1_unix.go:227 +0x353
runtime.goexit()
	/usr/lib/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 7 [chan receive]:
github.com/odeke-em/drive/src.(*Commands).Push.func1(0xc8200d62a0, 0xc820013fc0, 0xc820015040)
	/tmp/go/src/github.com/odeke-em/drive/src/push.go:52 +0x43
created by github.com/odeke-em/drive/src.(*Commands).Push
	/tmp/go/src/github.com/odeke-em/drive/src/push.go:56 +0x335

goroutine 19 [IO wait]:
net.runtime_pollWait(0x7f56bf1301a8, 0x72, 0xc82000e250)
	/usr/lib/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8200d87d0, 0x72, 0x0, 0x0)
	/usr/lib/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8200d87d0, 0x0, 0x0)
	/usr/lib/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8200d8770, 0xc820138000, 0x1000, 0x1000, 0x0, 0x7f56bf12a050, 0xc82000e250)
	/usr/lib/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820120008, 0xc820138000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/lib/go/src/net/net.go:172 +0xe4
crypto/tls.(*block).readFromUntil(0xc820112210, 0x7f56bc8e5028, 0xc820120008, 0x5, 0x0, 0x0)
	/usr/lib/go/src/crypto/tls/conn.go:455 +0xcc
crypto/tls.(*Conn).readRecord(0xc82012c000, 0x9ead17, 0x0, 0x0)
	/usr/lib/go/src/crypto/tls/conn.go:540 +0x2d1
crypto/tls.(*Conn).Read(0xc82012c000, 0xc8203fe000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/lib/go/src/crypto/tls/conn.go:901 +0x167
net/http.noteEOFReader.Read(0x7f56bc8ed9c8, 0xc82012c000, 0xc82000a3c8, 0xc8203fe000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/lib/go/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc8203fc000, 0xc8203fe000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	<autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc8203a8c60)
	/usr/lib/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc8203a8c60, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/lib/go/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc82000a370)
	/usr/lib/go/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
	/usr/lib/go/src/net/http/transport.go:685 +0xc78

goroutine 20 [select]:
net/http.(*persistConn).writeLoop(0xc82000a370)
	/usr/lib/go/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
	/usr/lib/go/src/net/http/transport.go:686 +0xc9d

goroutine 25 [IO wait]:
net.runtime_pollWait(0x7f56bf1300e8, 0x72, 0xc82000e250)
	/usr/lib/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8200d88b0, 0x72, 0x0, 0x0)
	/usr/lib/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8200d88b0, 0x0, 0x0)
	/usr/lib/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8200d8850, 0xc820420000, 0x1000, 0x1000, 0x0, 0x7f56bf12a050, 0xc82000e250)
	/usr/lib/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8200323b0, 0xc820420000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/lib/go/src/net/net.go:172 +0xe4
crypto/tls.(*block).readFromUntil(0xc820100b70, 0x7f56bc8e5028, 0xc8200323b0, 0x5, 0x0, 0x0)
	/usr/lib/go/src/crypto/tls/conn.go:455 +0xcc
crypto/tls.(*Conn).readRecord(0xc82000d080, 0x9ead17, 0x0, 0x0)
	/usr/lib/go/src/crypto/tls/conn.go:540 +0x2d1
crypto/tls.(*Conn).Read(0xc82000d080, 0xc820442000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/lib/go/src/crypto/tls/conn.go:901 +0x167
net/http.noteEOFReader.Read(0x7f56bc8ed9c8, 0xc82000d080, 0xc8203da738, 0xc820442000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/lib/go/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc82042db80, 0xc820442000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	<autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc8203a99e0)
	/usr/lib/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc8203a99e0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/lib/go/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc8203da6e0)
	/usr/lib/go/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
	/usr/lib/go/src/net/http/transport.go:685 +0xc78

goroutine 26 [select]:
net/http.(*persistConn).writeLoop(0xc8203da6e0)
	/usr/lib/go/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
	/usr/lib/go/src/net/http/transport.go:686 +0xc9d
```

--- a/src/remote.go
+++ b/src/remote.go
@@ -291,6 +291,10 @@ func (r *Remote) FindByPathTrashedM(p string) chan *File {
 }
 
 func reqDoPage(req *drive.FilesListCall, hidden bool, promptOnPagination bool) chan *File {
+	return _reqDoPage(req, hidden, promptOnPagination, false)
+}
+
+func _reqDoPage(req *drive.FilesListCall, hidden bool, promptOnPagination, nilOnNoMatch bool) chan *File {
 	fileChan := make(chan *File)
 
 	throttle := time.Tick(1e7)
@@ -319,7 +323,7 @@ func reqDoPage(req *drive.FilesListCall, hidden bool, promptOnPagination bool) c
 			}
 			pageToken = results.NextPageToken
 			if pageToken == "" {
-				if len(results.Items) < 1 && pageIterCount < 1 {
+				if nilOnNoMatch && len(results.Items) < 1 && pageIterCount < 1 {
 					// Item absolutely doesn't exist
 					fileChan <- nil
 				}
@@ -937,7 +941,7 @@ func (r *Remote) findByPathRecvRawM(parentId string, p []string, trashed bool) c
 		}
 		req.Q(expr)
 
-		resultsChan := reqDoPage(req, true, false)
+		resultsChan := _reqDoPage(req, true, false, true)
 		if len(rest) < 1 {
 			chanOChan <- resultsChan
 			return

